beaker/src/model_cache.rs

```diff
@@ -41,22 +46,122 @@ fn verify_checksum(path: &Path, expected_md5: &str) -> Result<bool> {
     Ok(actual_md5 == expected_md5)
 }
 
+/// Get detailed file information for debugging
+fn get_file_info(path: &Path) -> Result<String> {
+    let metadata = fs::metadata(path)?;
+    let size = metadata.len();
+    let size_mb = size as f64 / (1024.0 * 1024.0);
+
+    Ok(format!(
+        "size: {} bytes ({:.2} MB), permissions: {:?}, modified: {:?}",
+        size,
+        size_mb,
+        metadata.permissions(),
+        metadata
+            .modified()
+            .unwrap_or(std::time::SystemTime::UNIX_EPOCH)
+    ))
+}
+
 /// Download a model from URL to the specified path
 fn download_model(url: &str, output_path: &Path) -> Result<()> {
     log::info!("📥 Downloading model from: {url}");
 
     // Create parent directory if it doesn't exist
     if let Some(parent) = output_path.parent() {
         fs::create_dir_all(parent)?;
+        log::debug!("📁 Created parent directory: {}", parent.display());
+    }
+
+    // Check available disk space before download
+    if let Ok(parent) = output_path
+        .parent()
+        .ok_or_else(|| anyhow!("No parent directory"))
+    {
+        if let Ok(_metadata) = fs::metadata(parent) {
+            log::debug!("📊 Parent directory exists, checking disk space...");
+        }
+    }
+
+    // Download the file with better error handling and redirect support
+    log::debug!("🌐 Starting HTTP request...");
+    let client = reqwest::blocking::Client::builder()
+        .redirect(reqwest::redirect::Policy::limited(10))
+        .build()?;
+
+    let response = client
+        .get(url)
+        .send()
+        .map_err(|e| anyhow!("Failed to send HTTP request: {}", e))?;
+
+    let status = response.status();
+    log::debug!("📡 HTTP response status: {status}");
+
+    if !status.is_success() {
+        return Err(anyhow!("HTTP request failed with status: {}", status));
+    }
+
+    let content_length = response.content_length();
+
+    if let Some(length) = content_length {
+        let length_mb = length as f64 / (1024.0 * 1024.0);
+        log::info!("📏 Expected download size: {length} bytes ({length_mb:.2} MB)");
```

Comment:
This and all new logging outside of an error path should be debug level. Looking when we’ve found a problem should be warn or error.